jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/WorkflowProcessImpl.java

```diff
@@ -116,4 +116,30 @@ protected void processNodeContainer(org.jbpm.workflow.core.NodeContainer nodeCon
             }
         }
     }
+
+    protected void getParentNodes(Node currentNode, org.jbpm.workflow.core.NodeContainer nodeContainer, List<Node> nodes, long nodeId) {
```

Comment:
why this is not returning found node and instead populates the list? Asking as it seems this just stops when node is found so wouldn't it be better to return found node so the list won't be needed at all?

Reply:
Yes I changed the function a bit to stop when it find a node that accomplish the condition and return that node and the list is not needed at all :)